Source/WebCore/Modules/mediastream/libwebrtc/LibWebRTCPeerConnectionBackend.cpp

```diff
@@ -331,9 +331,9 @@ ExceptionOr<Ref<RTCRtpSender>> LibWebRTCPeerConnectionBackend::addTrack(MediaStr
 }
 
 template<typename T>
-ExceptionOr<Ref<RTCRtpTransceiver>> LibWebRTCPeerConnectionBackend::addTransceiverFromTrackOrKind(T&& trackOrKind, const RTCRtpTransceiverInit& init)
+ExceptionOr<Ref<RTCRtpTransceiver>> LibWebRTCPeerConnectionBackend::addTransceiverFromTrackOrKind(T&& trackOrKind, const RTCRtpTransceiverInit& init, IgnoreNegotiationNeededFlag ignoreNegotiationNeededFlag)
```

Comment:
Let's instead ASSERT that ignoreNegotiationNeededFlag is `No`.
I do not think this works for libwebrtc since the negotiation needed flag might be triggered asynchronously, for instance after the chaining of operations.

Reply:
> I do not think this works for libwebrtc since the negotiation needed flag might be triggered asynchronously, for instance after the chaining of operations.

I have verified that `LibWebRTCMediaEndpoint::OnNegotiationNeededEvent()` returns early when legacy offer options are processed, using the layout test `http/wpt/webrtc/RTCPeerConnection-createOffer-offerToReceive.html`.

I also tried a simpler approach, by setting the libwebrtc legacy offer options directly in `LibWebRTCMediaEndpoint::doCreateOffer()` without prior management in the top-level RTCPeerConnection, but that triggered various errors in the layout test.

I think the current approach of this PR works as expected :)